app/models/piece.rb

```diff
@@ -1,53 +1,93 @@
 class Piece < ActiveRecord::Base
   belongs_to :game
 
-  # Returns true/false if a piece is obstructed in its movement.
-  # Errors for movements not diagonal, vertical or horizontal
+  def move_to!(x, y)
+  # Immediately return to controller with false if move is deemed invalid
+    return false if !valid_move? x, y
```

Comment:
Does this make sense for all pieces? Does `valid_move?` return as we expect for, say, the knight?